### PR TITLE
OPAL-2915 - cascade delete all project's vcfSampleMapping when it is …

### PR DESCRIPTION
**Pull request by meek0**
…deleted

### CODE REVIEW

File: opal-core/src/main/java/org/obiba/opal/core/service/VCFSamplesMappingServiceImpl.java [hunk 1]
```diff
@@ -70,6 +70,15 @@ public void delete(@NotNull String projectName) throws NoSuchVCFSamplesMappingEx
   }
 
   @Override
+  public void deleteProjectSampleMappings(@NotNull String project) {
+    Iterable<VCFSamplesMapping> list = orientDbService.list(
+        VCFSamplesMapping.class,
+        "select from " + VCFSamplesMapping.class.getSimpleName() + " where tableReference like ?",
+        project + ".%");
+    list.forEach(s -> delete(s.getProjectName()));
```
Comment (ymarcon) [inline opal-core/src/main/java/org/obiba/opal/core/service/VCFSamplesMappingServiceImpl.java]: why not using findUnique() like in getVCFSamplesMapping()? Do you expect to have multiple mappings for one project?

Comment (meek0) [inline opal-core/src/main/java/org/obiba/opal/core/service/VCFSamplesMappingServiceImpl.java]: The same mapping table can be used in multiple projects.

Comment (ymarcon) [inline opal-core/src/main/java/org/obiba/opal/core/service/VCFSamplesMappingServiceImpl.java]: Oh ok good, you delete mapping when it is used by the other projects as well.
And what if the table is deleted all alone?

Comment (meek0) [inline opal-core/src/main/java/org/obiba/opal/core/service/VCFSamplesMappingServiceImpl.java]: Ah yes, I will add a cascade for this as well.

**Pull request merged**